### PR TITLE
Log save/load in teehistorian

### PR DESCRIPTION
**Pull request by Zwelf**
Including tests

### CODE REVIEW

File: src/game/server/gamecontext.h [hunk 1]
```diff
@@ -126,6 +126,8 @@ class CGameContext : public IGameServer
 	CTuningParams *Tuning() { return &m_Tuning; }
 	CTuningParams *TuningList() { return &m_aTuningList[0]; }
 	IAntibot *Antibot() { return m_pAntibot; }
+	CTeeHistorian* TeeHistorian() { return &m_TeeHistorian; }
```
Comment (heinrich5991) [inline src/game/server/gamecontext.h:129]:
```suggestion
	CTeeHistorian *TeeHistorian() { return &m_TeeHistorian; }
```

File: src/game/server/teehistorian.cpp [hunk 2]
```diff
@@ -4,6 +4,9 @@
 #include <engine/shared/snapshot.h>
 #include <engine/shared/json.h>
 #include <game/gamecore.h>
+#if defined(CONF_SQL)
+#include "score/sql_score.h"
+#endif
```
Comment (heinrich5991) [inline src/game/server/teehistorian.cpp:9]: Teehistorian having a dependency on SQL stuff isn't really nice. Could go with it if the code gets worse without it. This makes the source file less portable (across e.g. Teeworlds mod codebases).

File: src/game/server/teehistorian.cpp [hunk 3]
```diff
@@ -490,6 +493,43 @@ void CTeeHistorian::RecordTestExtra()
 	WriteExtra(UUID_TEEHISTORIAN_TEST, "", 0);
 }
 
+#if defined(CONF_SQL)
```
Comment (heinrich5991) [inline src/game/server/teehistorian.cpp:496]: Conditional compilation is annoying, if there's a way without it, I'd strongly prefer removing it. Tests would need to be run for every conditional compilation option, which isn't feasible.

File: src/game/server/teehistorian.cpp [hunk 4]
```diff
@@ -490,6 +493,43 @@ void CTeeHistorian::RecordTestExtra()
...
+		TeehistorianID = UUID_TEEHISTORIAN_SAVE_FAILURE;
+		break;
+	case CSqlSaveResult::LOAD_SUCCESS:
+		Buffer.AddRaw(&pSaveResult->m_SaveID, sizeof(pSaveResult->m_SaveID));
+		Buffer.AddString(pSaveResult->m_SavedTeam.GetString(), 0);
+		TeehistorianID = UUID_TEEHISTORIAN_LOAD_SUCCESS;
+		break;
+	case CSqlSaveResult::LOAD_FAILED:
+		TeehistorianID = UUID_TEEHISTORIAN_LOAD_FAILURE;
+		break;
```
Comment (heinrich5991) [inline src/game/server/teehistorian.cpp:521]: Missing default case with `dbg_assert(false, "unreachable");`

File: src/game/server/teehistorian.cpp [hunk 5]
```diff
@@ -490,6 +493,43 @@ void CTeeHistorian::RecordTestExtra()
...
+	}
+
+	if(m_Debug)
+	{
+		dbg_msg("teehistorian", "save_load team=%d status=%d", Team, pSaveResult->m_Status);
```
Comment (heinrich5991) [inline src/game/server/teehistorian.cpp:526]: Missing debug output with all the information that is put into the resulting teehistorian.

File: src/game/server/gamecontext.h (hunk 1 above)
Comment (Zwelf) [reply src/game/server/gamecontext.h]: Fixed

File: src/game/server/teehistorian.cpp (hunk 2 above)
Comment (Zwelf) [reply src/game/server/teehistorian.cpp]: Now it isn't even dependent on `score.h`

**Comment by Zwelf** [comment]
Rebased on master and integrated feedback into commits.

File: src/game/server/teehistorian.cpp [hunk 6]
```diff
@@ -490,6 +490,70 @@ void CTeeHistorian::RecordTestExtra()
...
+	Buffer.Reset();
+	Buffer.AddInt(Team);
+	Buffer.AddRaw(&SaveID, sizeof(SaveID));
+	Buffer.AddString(pTeamSave, 0);
+
+	if(m_Debug)
+	{
+		char aSaveID[UUID_MAXSTRSIZE];
+		FormatUuid(SaveID, aSaveID, sizeof(aSaveID));
+		dbg_msg("teehistorian", "save_load team=%d save_id=%s team_save='%s'", Team, aSaveID, pTeamSave);
```
Comment (heinrich5991) [inline src/game/server/teehistorian.cpp:505]: Here and in the corresponding places: `save_success` etc.

**Pull request merged**